src/test/WstETHAdapterV1.t.sol

```diff
@@ -25,50 +35,39 @@ contract WstETHAdapterV1Test is DSTestPlus {
-            alchemist:       address(this),
+            alchemist:       address(alchemist),
             token:           address(wstETH),
             parentToken:     address(stETH),
             underlyingToken: address(weth),
             curvePool:       address(curvePool),
+            oracleStethUsd:  address(oracleStethUsd),
+            oracleEthUsd:    address(oracleEthUsd),
             ethPoolIndex:    0,
             stEthPoolIndex:  1,
             referral:        address(0)
         }));
-    }
-
-    function testPrice() external {
-        uint256 decimals = SafeERC20.expectDecimals(address(wstETH));
-        assertEq(adapter.price(), stETH.getPooledEthByShares(10**decimals));
-    }
-
-    function testWrap() external {
-        tip(address(weth), address(this), 1e18);
-
-        SafeERC20.safeApprove(address(weth), address(adapter), 1e18);
-        uint256 wrapped = adapter.wrap(1e18, address(0xbeef));
 
-        assertEq(weth.allowance(address(this), address(adapter)), 0);
-        assertEq(wstETH.balanceOf(address(0xbeef)), wrapped);
+        hevm.startPrank(admin);
+        IWhitelist(whitelistETHAddress).add(address(this));
+        alchemist.setMaximumExpectedValue(address(wstETH), 1000000000e18);
+        hevm.stopPrank();
     }
 
-    function testUnwrap() external {
-        tip(address(wstETH), address(this), 1e18);
-
-        SafeERC20.safeApprove(address(wstETH), address(adapter), 1e18);
-        uint256 unwrapped = adapter.unwrap(1e18, address(0xbeef));
-
-        assertEq(wstETH.allowance(address(this), address(adapter)), 0);
-        assertEq(weth.balanceOf(address(0xbeef)), unwrapped);
-    }
-
-    function testWrapGrief() external {
-        tip(address(weth), address(this), 1e18);
-        tip(address(wstETH), address(adapter), 1e18);
-
-        SafeERC20.safeApprove(address(weth), address(adapter), 1e18);
-        uint256 wrapped = adapter.wrap(1e18, address(0xbeef));
-
-        assertEq(weth.allowance(address(this), address(adapter)), 0);
-        assertEq(wstETH.balanceOf(address(0xbeef)), wrapped);
+    function testRoundTrip() external {
+        deal(address(weth), address(this), 1e18);
+
+        SafeERC20.safeApprove(address(weth), address(alchemist), 1e18);
+        uint256 shares = alchemist.depositUnderlying(address(wstETH), 1e18, address(this), 0);
+
+        // Test that price function returns value within 0.1% of actual
+        uint256 underlyingValue = shares * adapter.price() / 10**SafeERC20.expectDecimals(address(wstETH));
+        assertGt(underlyingValue, 1e18 * 9990 / BPS);
+        
+        SafeERC20.safeApprove(adapter.token(), address(adapter), shares);
+        uint256 unwrapped = alchemist.withdrawUnderlying(address(wstETH), shares, address(this), shares * 9990 / 10000);
+        
+        assertEq(weth.balanceOf(address(this)), unwrapped);
+        assertEq(wstETH.balanceOf(address(this)), 0);
+        assertEq(wstETH.balanceOf(address(adapter)), 0);
```

Comment:
also check wsteth balance of alchemist is 0 (since the alchemist is the one holding the yield tokens)